projects/js-packages/charts/src/components/pie-chart/pie-chart.tsx

```diff
@@ -243,21 +243,42 @@ const PieChartInternal = ( {
 										pathProps.onMouseLeave = onMouseLeave;
 									}
 
+									// Estimate text width more accurately for background sizing
+									const fontSize = 12;
+									const estimatedTextWidth = arc.data.label.length * fontSize * 0.6; // Rough estimate
```

Comment:
Could we please try to use `getStringWidth` insteand: https://github.com/Automattic/jetpack/blob/2cf87570ee3df0e6063c05e442bde8ad9803a137/projects/js-packages/charts/src/components/shared/utils.ts#L24